graylog2-server/src/main/java/org/graylog/plugins/map/geoip/GeoIpResolverFactory.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+package org.graylog.plugins.map.geoip;
+
+import com.codahale.metrics.Timer;
+import org.graylog.plugins.map.config.DatabaseType;
+import org.graylog.plugins.map.config.DatabaseVendorType;
+import org.graylog.plugins.map.config.GeoIpResolverConfig;
+
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * A factory to create ASN and Location {@link GeoIpResolver} resolvers based on the {@link DatabaseVendorType} contained in
+ * the current {@link GeoIpResolverConfig}.
+ */
+public class GeoIpResolverFactory {
```

Comment:
May have found a way to create a proper Guice factory.  Will try it out after break.